platform/platform_freebsd.cpp

```diff
@@ -0,0 +1,142 @@
+#include "platform/platform.hpp"
```

Comment:
> I agree that we don’t need runtime polymorphism, but it’s just lesser evil.

By

> there's no need to use run-time polymorphism here

I've meant that there will be no runtime polymorphism in fact. With C++11 override & final keywords properly used, when final subclass is used directly (our case), compilers can do devirtualization and avoid vtbl lookups completely. There will not be any noticeable effect on performance here anyway.